### PR TITLE
[WIP] Change attribute `most_likely_phase` to `phase` in PhaseEstimator

### PR DESCRIPTION
**Pull request by jlapeyre**
IterativePhaseEstimation #6096 will satisfy the PhaseEstimator interface.
However, the property for retrieving the phase is currently
`PhaseEstimator.most_likely_phase`, which is too specific to accomodate
IterativePhaseEstimation, which only computes a single phase in any
case. This PR renames this property to `phase`, and leaves the
definition of `phase` to the specific implementation.

This PR currently does not include a deprecated version of the previous attribute `most_likely_phase`. Perhaps it should.

### CODE REVIEW

**Comment by Cryoris** [comment]
> This PR currently does not include a deprecated version of the previous attribute most_likely_phase. Perhaps it should.

Yes, I think it should, could you add one? 🙂

File: qiskit/algorithms/phase_estimators/hamiltonian_phase_estimation_result.py [hunk 1]
```diff
@@ -81,6 +82,17 @@ def filter_phases(self, cutoff: float = 0.0, scaled: bool = True,
         else:
             return cast(Dict, phases)
 
+    @property
+    @deprecate_function('The HamiltonianPhaseEstimationResult.most_likely_phase attribute '
```
Comment (Cryoris) [inline qiskit/algorithms/phase_estimators/hamiltonian_phase_estimation_result.py:86]: Could you use our standard text to include when it's been deprecated etc.? 🙂 
```python
import warnings

def foo(input):
    warnings.warn('The qiskit.foo() function is deprecated as of 0.9.0, and '
                  'will be removed no earlier than 3 months after that '
                  'release date. You should use the qiskit.bar() function '
                  'instead.', DeprecationWarning, stacklevel=2)

```
(see https://qiskit.org/documentation/contributing_to_qiskit.html#deprecation-warnings)

Comment (jlapeyre) [reply qiskit/algorithms/phase_estimators/hamiltonian_phase_estimation_result.py:86]: Ok. Btw, I modeled this on deprecated functions in quantumcircuit.py. According to the doc you link above, `qiskit.utils.deprecation` should never be used?

**Review by Cryoris — APPROVED** [review]
LGTM 👍🏻

**Comment by woodsp-ibm** [comment]
I can't help feel we ought to remove [WIP] from titles before merging. The other alternative is to have them as Draft that way there is no forgetting to change the title since as a Draft I would assume it would not need [WIP] too!

**Pull request merged**